### PR TITLE
refactor(yaml): cleanup `stringifyFlowMapping()`

### PR DESCRIPTION
**Pull request by timreichen**

### CODE REVIEW

**Review by iuioiua — COMMENTED** [review]
This method is not yet fully tested. I'd only feel completely comfortable merging this PR once it's fully tested. See https://app.codecov.io/gh/denoland/std/blob/main/yaml%2F_dumper_state.ts#L627

File: yaml/_dumper_state.ts [hunk 1]
```diff
@@ -628,52 +628,41 @@ export class DumperState {
...
-        continue; // Skip this pair because of invalid value.
-      }
+      let pairBuffer = "";
 
+      pairBuffer += quote;
+      if (keyString.length > 1024) pairBuffer += "? ";
+      pairBuffer += keyString;
+      pairBuffer += quote;
+      pairBuffer += separator;
       pairBuffer += valueString;
```
Comment (iuioiua) [inline yaml/_dumper_state.ts:660]: Can be done in one line.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Frankly, I don't really see the purpose behind this PR.

**Comment by timreichen** [comment]
> Frankly, I don't really see the purpose behind this PR.

There are several (tiny) performance benefits:
- `quote` and `separator` are take out of the loop, therefore only need one check
- `stringifyNode()`for key and value is called at the beginning of the loop, thus continues the loop before `pairBuffer` is even initialized.

It also is more readable imo.

**Review by iuioiua — APPROVED** [review]
Thank you

**Pull request merged**